Dess wrd templates (#1870)
telerik/reporting-docs: designing-reports/report-designer-tools/web-report-designer/configuring-report-templates.md

````CSharp
services.TryAddSingleton<IReportDesignerServiceConfiguration>(sp => new ReportDesignerServiceConfiguration
{
	TemplateDefinitionStorage = new FileTemplateDefinitionStorage("templates_folder_path", new[] { "sub_folder_to_exclude" }),
});
````

>note Report templates (.trtx files) can be stored only in the configured `Report Templates` folder. Other file formats are not allowed in the folder.

## Managing Permissions

To restrict specific actions or features for users on the client side, for example to restrict `Report Templates`, the [DeniedPermissions]({%slug telerikreporting/report-designer-tools/web-report-designer/web-report-designer-customization%}) setting of the `ReportDesignerServiceConfiguration` can be used. It allows developers to customize the designer experience by disabling certain tools, components, or capabilities based on application logic or user roles.

> >note You can override the [GetDeniedPermissions]({%slug telerikreporting/report-designer-tools/web-report-designer/web-report-designer-customization%}) method in your custom implementation of the reporting service to apply different restrictions based on the logged-in user or other conditions.

Reviewer: ```suggestion
>note You can override the [`GetDeniedPermissions`]({%slug telerikreporting/report-designer-tools/web-report-designer/web-report-designer-customization%}) method in your custom implementation of the reporting service and apply different restrictions based on the currently logged-in user or other conditions.
```

### Restricting the Templates Folder

````CSharp
services.TryAddSingleton<IReportDesignerServiceConfiguration>(sp => new ReportDesignerServiceConfiguration
{
    DeniedPermissions = ReportDesignerPermissionsBuilder.Build(
        Permission.Commands_AssetsManager_ReportTemplates
        )
}));
````

> Once the end-user is **not** granted permissions for the Report Templates, the following actions are available:

Reviewer: ```suggestion
Users with restriced access can still perform some operations:
```

* Create a New Report from a Template - The **From Template** option remains visible even when the user lacks permissions to manage templates. In this restricted mode:

  * Users **cannot** create or upload new templates.
  * Users **cannot** delete or edit existing templates.
  * Users **can** still create new reports using the available templates.

>caption Restricted User

![Restricted User From Template ><](images/web-report-designer-report-templates-restricted-user.png)

>caption Default User

![Allowed User From Template ><](images/web-report-designer-report-templates-allowed-user.png)

> * Opening the **Assets manager** - When a *restricted* user opens the Assets Manager, the `Report Templates` folder is invisible and inaccessible:

Reviewer: ```suggestion
* Open the **Assets manager**&mdash;When a restricted user opens the Assets Manager, the `Report Templates` folder is invisible and inaccessible:
```

|Restricted User|Default User|
|----|----|
|![Restricted User Assets Manager ><](images/web-report-designer-report-templates-restricted-user-assets-manager.png)|![Allowed User Assets Manager ><](images/web-report-designer-report-templates-allowed-user-assets-manager.png)|     

> * **Save Report As** - The *restricted* end-user can **only** save as a **Report**. The option for saving as a **Template** is hidden. 

Reviewer: ```suggestion
* **Save Report As**&mdash;The restricted end-user can save the new report only as a **Report**. The option for saving as a **Template** is hidden. 
```
